### PR TITLE
chore: Add tenderly simulation and data verification for `setProductTypes`

### PR DESCRIPTION
**Pull request by MilGard91**
## Context

Missing simluation tx and data verification of the transaction

## Changes proposed in this pull request

Add simulation tx for tenderly and verify the data

## Test plan

Please describe the tests cases that you ran to verify your changes. Add further instructions on 
how to run them if needed (i.e. migration / deployment scripts, env vars, etc).

## Checklist

- [x] Rebased the base branch
- [ ] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: scripts/products/get-setProductTypes-txdata.js [hunk 1]
```diff
@@ -4,11 +4,52 @@ const fs = require('fs');
...
+    if (!decodedProductType) {
+      console.log(`\nVerification failed. No matching product found for: ${productTypeName}`);
+      return;
+    }
+
+    const { claimMethod, gracePeriod } = decodedProductType.productType;
+    //
+    // Set csv default values
+    const csvProductTypeId = csvProductTypes.Id || MaxUint256.toString();
+    const csvGracePeriod = csvProductTypes['Grace Period (days)'] * 24 * 3600; // This MUST be in seconds
```
Comment (rackstar) [inline scripts/products/get-setProductTypes-txdata.js:32]: nit: should we explicitly convert to Number?

File: scripts/products/get-setProductTypes-txdata.js [hunk 2]
```diff
@@ -4,11 +4,52 @@ const fs = require('fs');
...
+    console.log(decodedTxInputs);
+    const decodedProductType = decodedTxInputs.find(decoded => decoded.productTypeName === productTypeName);
+
+    if (!decodedProductType) {
+      console.log(`\nVerification failed. No matching product found for: ${productTypeName}`);
+      return;
+    }
+
+    const { claimMethod, gracePeriod } = decodedProductType.productType;
+    //
```
Comment (rackstar) [inline scripts/products/get-setProductTypes-txdata.js:29]: nit: remove empty comment block

File: scripts/products/get-setProductTypes-txdata.js [hunk 3]
```diff
@@ -4,11 +4,52 @@ const fs = require('fs');
...
+const { COVER_ADDRESS, IPFS_API_URL } = constants;
+
+const ipfs = ipfsClient({ url: IPFS_API_URL });
+
+const verifyDecodedTxInputs = (csvProductTypeData, decodedTxInputs) => {
+  for (const csvProductTypes of csvProductTypeData) {
+    // Find matching object in decodedTxInputs by 'Product Name'
+    const productTypeName = csvProductTypes.Name;
+    console.log(productTypeName);
+    console.log(decodedTxInputs);
```
Comment (rackstar) [inline scripts/products/get-setProductTypes-txdata.js:20]: is this left over dev logs?

File: scripts/products/get-setProductTypes-txdata.js (hunk 1 above)
Comment (MilGard91) [reply scripts/products/get-setProductTypes-txdata.js:32]: It does it automatically when the operation is multiplication, but I'll add it just for cleaner code

**Review by rackstar — APPROVED** [review]

**Pull request merged**